sklearn/preprocessing/_data.py

```diff
@@ -3428,9 +3428,21 @@ def inverse_transform(self, X):
             "yeo-johnson": self._yeo_johnson_inverse_transform,
         }[self.method]
         for i, lmbda in enumerate(self.lambdas_):
-            with np.errstate(invalid="ignore"):  # hide NaN warnings
-                X[:, i] = inv_fun(X[:, i], lmbda)
-
+            with warnings.catch_warnings(record=True) as captured_warnings:
+                with np.errstate(invalid="warn"):
+                    X[:, i] = inv_fun(X[:, i], lmbda)
+
+            if captured_warnings:
+                last_warning = captured_warnings[-1]
```

Comment:
Should we check all the warnings and not just the last one?

```python
if any("invalid value encountered in power" in str(w.message) for w in captured_warnings):
    ...

Reply:
Ok, done.